File: examples/movie/MovieData_5_Mapper.java
// Movie Data 5 Mapper

public class Mapper {
    public void map(String key, String value) {
        String data[] = key.split(",");
        String rating = data[2];
        int date = Integer.parseInt(data[3]);
        int secsInYear = 31556926, startYear = 1970, july1 = 15638400, july31 = 18316799;
        int year = (int) (((double) date / (double) secsInYear) + startYear);
        int diff = year - startYear, numOfYears = secsInYear * diff;
        int julyStart = july1 + numOfYears, julyEnd = july31 + numOfYears;
        if((date >= july1) && (date <= july31)) {
            Wmr.emit(Integer.toString(year), rating);
        }
        else if((date >= julyStart) && (date <= julyEnd)) {
            Wmr.emit(Integer.toString(year), rating);
        }
    }
}

File: examples/MovieData_5_Mapper.py
# Movie Data 5 Mapper

def mapper(key, value):
    secondsInYear = 31556926
    startYear = 1970
    july1 = 15638400
    july31 = 18316799
    data = key.split(',')
    rating = data[2]
    date = float(data[3])
    year = int((date / secondsInYear) + startYear)
    diff = year - startYear
    numOfYears = secondsInYear * diff
    julyStart = july1 + numOfYears
    julyEnd = july31 + numOfYears
    if ((date >= july1) and (date <= july31)):
        Wmr.emit(year, rating)
    elif ((date >= julyStart) and (date <= julyEnd)):
        Wmr.emit(year, rating)